pkg/controller/pv.go

```diff
@@ -101,17 +97,23 @@ func (r *reconcilePV) Reconcile(ctx context.Context, request reconcile.Request)
 		}
 
 		klog.V(5).Infof("Max Retry (%d) for Get AzVolume (%s) exceeded. The CRI is probably deleted.", maxRetry, azVolumeName)
-		r.retryMutex.Lock()
-		delete(r.retryMap, azVolumeName)
-		r.retryMutex.Unlock()
+		r.retryMap.Delete(azVolumeName)
 		return reconcile.Result{}, nil
 	}
 
 	if azVolume.Status.Detail != nil {
-		if pv.Status.Phase == corev1.VolumeReleased && azVolume.Status.Detail.Phase == v1alpha1.VolumeBound {
-			updated := azVolume.DeepCopy()
+		updated := azVolume.DeepCopy()
+		switch phase := pv.Status.Phase; phase {
+		case corev1.VolumeBound:
+			pvClaimName := getNamespacedName(pv.Spec.ClaimRef.Namespace, pv.Spec.ClaimRef.Name)
+			updated.Status.Detail.Phase = v1alpha1.VolumeBound
+			r.controllerSharedState.volumeToClaimMap.Store(azVolumeName, pvClaimName)
+			r.controllerSharedState.claimToVolumeMap.Store(pvClaimName, azVolumeName)
```

Comment:
Consider making lines 110-111 a method of `SharedState`, e.g. `addVolumeClaimMapping`.